js/app.js

```diff
@@ -97,9 +151,27 @@ function createDefaultSnippetPreview() {
  * @constructor
  */
 YoastSEO.App = function( args ) {
-	this.config = this.extendConfig( args );
+	if ( !isObject( args ) ) {
+		args = {};
+	}
+	defaultsDeep( args, defaults );
+
+	this.config = args;
 	this.callbacks = this.config.callbacks;
 
+	if ( !isObject( this.callbacks.getData ) ) {
```

Comment:
I would do these checks before we're even assigning them to our `App` properties.